packages/devtools-client-adapters/src/types.js

```diff
@@ -0,0 +1,162 @@
+// @flow
+/**
+ * These are intended to be generic types used by Firefox, Chrome, Node, and
+ * possibly others.
+ *
+ * They can be imported into clients by using the following Flow import:
+ * `import type { Location } from 'devtools-client-adapters'`
+ */
+
+/**
+ * Breakpoint ID
+ *
+ * @memberof types
+ * @static
+ */
+export type BreakpointId = string;
+
+/**
+ * Source ID
+ *
+ * @memberof types
+ * @static
+ */
+export type SourceId = string;
+
+/**
+ * Actor ID
+ *
+ * @memberof types
+ * @static
+ */
+export type ActorId = string;
+
+/**
+ * Source File Location
+ *
+ * @memberof types
+ * @static
+ */
+export type Location = {
+  sourceId: SourceId,
+  line: number,
+  column?: number
+};
+
+/**
+ * Breakpoint
+ *
+ * @memberof types
+ * @static
+ */
+export type Breakpoint = {
+  id: BreakpointId,
+  location: Location,
+  loading: boolean,
+  disabled: boolean,
+  text: string,
+  condition: ?string,
+};
+
+/**
+ * Breakpoint Result is the return from an add/modify Breakpoint request
+ *
+ * @memberof types
+ * @static
+ */
+export type BreakpointResult = {
+  id: ActorId,
+  actualLocation: Location
+};
+
+/**
+ * Source
+ *
+ * @memberof types
+ * @static
+ */
+export type Source = {
+  id: SourceId,
+  url?: string,
+  sourceMapURL?: string
+};
+
+/**
+ * Frame ID
+ *
+ * @memberof types
+ * @static
+ */
+export type FrameId = string;
```

Comment:
I pulled out a lot of these `id` types because we use them to reference things later so it felt better to have a specific identifier for each object so we could type the calls that look like `get(FrameId)`.